src/PixelFlags.cc

```diff
@@ -57,87 +59,153 @@ class FootprintBits : public afw::detection::FootprintFunctor<MaskedImageT> {
 private:
     typename MaskedImageT::Mask::Pixel _bits;
 };
-}  // end anonymous namespace
+
 PixelFlagsAlgorithm::PixelFlagsAlgorithm(
     Control const & ctrl,
     std::string const & name,
     afw::table::Schema & schema
-) : _ctrl(ctrl),
-    _centroidExtractor(schema, name)
+) : _ctrl(ctrl)
 {
-    static boost::array<FlagDefinition,N_FLAGS> const flagDefs = {{
-        {"flag", "general failure flag, set if anything went wrong"},
-        {"flag_edge", "Source is outside usable exposure region (masked EDGE or NO_DATA)"},
-        {"flag_interpolated", "Interpolated pixel in the source footprint"},
-        {"flag_interpolatedCenter", "Interpolated pixel in the source center"},
-        {"flag_saturated", "Saturated pixel in the source footprint"},
-        {"flag_saturatedCenter", "Saturated pixel in the source center"},
-        {"flag_cr", "Cosmic ray in the source footprint"},
-        {"flag_crCenter", "Cosmic ray in the source center"},
-        {"flag_bad", "Bad pixel in the source footprint"}
-    }};
-    _flagHandler = FlagHandler::addFields(schema, name, flagDefs.begin(), flagDefs.end());
+    // Add generic keys first, which don't correspond to specific mask planes
+    _generalFailure = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname,
+                                        "general failure flag, set if anything went wring");
+    _offImageKey = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_offimage", 
```

Comment:
Will be split out to a separate commit, when the code is cleaned up for final merge